values/object.go

```diff
@@ -87,43 +134,17 @@ func (o *object) String() string {
 }
 
 func (o *object) Type() semantic.MonoType {
-	// TODO (algow): finish implementation of object
-	return semantic.MonoType{}
+	return o.typ
 }
 
 func (o *object) Set(k string, v Value) {
```

Comment:
Should we rename this to `Replace` or something similar? Since it will not create a new property but instead will only replace the existing value. Can you add a comment describing its behavior?

Reply:
I think `Set` makes sense, but I'll add a godoc comment to the interface making it clear that it is only setting values and doesn't add anything.